### PR TITLE
Control: MRAC controller higher-order upgrade

### PR DESCRIPTION
**Pull request by kevin-y-wang**

### CODE REVIEW

File: modules/control/common/mrac_controller.cc [hunk 1]
```diff
@@ -19,23 +19,33 @@
...
 
-double MracController::Control(const double command, const double state,
+using Matrix = Eigen::MatrixXd;
+
+double MracController::Control(const double command, const Matrix state,
                                const double dt) {
-  // check if the reference model well set up during the initilization
+  // check if the reference/adaption model well set up during the initilization
   if (!reference_model_enabled_) {
-    TransformReferenceModel(dt);
```
Comment (Capri2014) [inline modules/control/common/mrac_controller.cc]: Don not build reference/adaption model in `::Control(...)`, it should only be done once in `Init()` and after that only update the model if needed.

Comment (kevin-y-wang) [reply modules/control/common/mrac_controller.cc]: Fixed per comment. Thanks

File: modules/control/common/mrac_controller.cc [hunk 2]
```diff
@@ -87,28 +98,25 @@ double MracController::Control(const double command, const double state,
...
-  state_reference_[1] = state_reference_[0];
-  state_action_[1] = state_action_[0];
-  input_desired_[1] = input_desired_[0];
+  state_reference_.col(1) = state_reference_.col(0);
+  state_action_.col(1) = state_action_.col(0);
+  input_desired_.col(1) = input_desired_.col(0);
   control_previous_ = control;
   return control;
-}
+}  // namespace control
```
Comment (sjiang2018) [inline modules/control/common/mrac_controller.cc]: remove // ....

File: modules/control/common/mrac_controller.cc (hunk 1 above)
Comment (Capri2014) [reply modules/control/common/mrac_controller.cc]: merge these two `if` please

Comment (kevin-y-wang) [reply modules/control/common/mrac_controller.cc]: Fixed

File: modules/control/common/mrac_controller.cc (hunk 2 above)
Comment (kevin-y-wang) [reply modules/control/common/mrac_controller.cc]: Thanks. Fixed per comment

**Comment by Capri2014** [comment]
1. Separate `Reset` to `Reset`, `ResetStates` and `ResetGain`, sometimes we may need only clear the internal state variables.

2. For the gain matrix(continuous one), it is always good to have the `set` functions as well in case we may need to adjust gains online in the future (i.e. change of scenarios).

**Comment by kevin-y-wang** [comment]
> 1. Separate `Reset` to `Reset`, `ResetStates` and `ResetGain`, sometimes we may need only clear the internal state variables.
> 2. For the gain matrix(continuous one), it is always good to have the `set` functions as well in case we may need to adjust gains online in the future (i.e. change of scenarios).

Refactored per comment. Thanks

**Pull request merged**